common/ast/src/parser/rule/statement.rs

```diff
@@ -47,3 +49,210 @@ pub fn statement(i: Input) -> IResult<Statement> {
+                    .unwrap_or_default(),
+                limit: limit_block.map(|(_, limit)| limit).unwrap_or_default(),
+            }
+        },
+    )(i)
+}
+
+pub fn select_target(i: Input) -> IResult<SelectTarget> {
+    let qualified_wildcard = map(
+        rule! {
+            ( #ident ~ "." ~ ( #ident ~ "." )? )? ~ "*"
+        },
+        |(res, _)| match res {
+            Some((fst, _, Some((snd, _)))) => SelectTarget::Indirections(vec![
+                Indirection::Identifier(fst),
+                Indirection::Identifier(snd),
+                Indirection::Star,
+            ]),
+            Some((fst, _, None)) => {
+                SelectTarget::Indirections(vec![Indirection::Identifier(fst), Indirection::Star])
+            }
+            None => SelectTarget::Indirections(vec![Indirection::Star]),
+        },
+    );
+    let projection = map(
+        rule! {
+            #expr ~ ( AS ~ #ident )?
+        },
+        |(expr, alias)| SelectTarget::Projection {
+            expr,
+            alias: alias.map(|(_, name)| name),
+        },
+    );
+
+    rule!(
+        #qualified_wildcard
+        | #projection
+    )(i)
+}
+
+pub fn aliased_table(i: Input) -> IResult<TableReference> {
+    let table_alias = map(rule! { #ident }, |name| TableAlias {
+        name,
+        columns: vec![],
+    });
+
+    map(
+        rule! {
+            #ident ~ ( "." ~ #ident )? ~ ( AS? ~ #table_alias )?
+        },
+        |(fst, snd, alias)| {
+            let (database, table) = match (fst, snd) {
+                (database, Some((_, table))) => (Some(database), table),
+                (table, None) => (None, table),
+            };
+
+            TableReference::Table {
+                database,
+                table,
+                alias: alias.map(|(_, alias)| alias),
+            }
+        },
+    )(i)
+}
+
+pub fn table_reference(i: Input) -> IResult<TableReference> {
+    rule! (
+        #parenthesized_joined_tables
+        | #aliased_table
+    )(i)
+}
+
+pub fn parenthesized_joined_tables(i: Input) -> IResult<TableReference> {
+    let with_parenthesis = map(
+        rule! {
+            "(" ~ #parenthesized_joined_tables ~ ")"
+        },
+        |(_, joined_tables, _)| joined_tables,
+    );
+
+    rule! (
+        # joined_tables
+        | #with_parenthesis
+    )(i)
+}
+
+pub fn joined_tables(i: Input) -> IResult<TableReference> {
+    let join_condition_on = map(
+        rule! {
+            ON ~ #expr
+        },
+        |(_, expr)| JoinCondition::On(expr),
+    );
+    let join_condition_using = map(
+        rule! {
+            USING ~ "(" ~ #comma_separated_list1(ident) ~ ")"
+        },
+        |(_, _, idents, _)| JoinCondition::Using(idents),
+    );
+    let join_condition = rule! { #join_condition_on | #join_condition_using };
+
+    let join = map(
+        rule! {
+            #aliased_table ~ #join_operator? ~ JOIN ~ #cut(table_reference) ~ #cut(join_condition)
```

Comment:
Maybe the `#aliased_table` should be a `table_reference`?

Reply:
addressed